diff --git a/src/src/CMakeLists.txt b/src/src/CMakeLists.txt
index d7f387c2..0cc2b01d 100644
--- a/src/src/CMakeLists.txt
+++ b/src/src/CMakeLists.txt
@@ -14,8 +14,10 @@ set(Uri_SRCS
   ${CMAKE_CURRENT_SOURCE_DIR}/detail/uri_resolve.cpp
   )
 
+find_package(Boost REQUIRED)
 add_library(network-uri ${Uri_SRCS})
-target_link_libraries(network-uri)
+target_compile_definitions(network-uri PRIVATE -DNETWORK_URI_EXTERNAL_BOOST)
+target_link_libraries(network-uri Boost::boost)
 if(${CMAKE_CXX_COMPILER_ID} MATCHES Clang)
   if (NOT Uri_DISABLE_LIBCXX)
     target_link_libraries(network-uri "c++")
diff --git a/src/src/detail/uri_normalize.cpp b/src/src/detail/uri_normalize.cpp
index 77452236..ed4c2ae7 100644
--- a/src/src/detail/uri_normalize.cpp
+++ b/src/src/detail/uri_normalize.cpp
@@ -10,6 +10,7 @@
 #ifdef NETWORK_URI_EXTERNAL_BOOST
 #include <boost/algorithm/string/split.hpp>
 #include <boost/algorithm/string/join.hpp>
+namespace network_boost = boost;
 #else   // NETWORK_URI_EXTERNAL_BOOST
 #include "../boost/algorithm/string/split.hpp"
 #include "../boost/algorithm/string/join.hpp"
diff --git a/src/src/detail/uri_resolve.cpp b/src/src/detail/uri_resolve.cpp
index 17fc68d8..c8882690 100644
--- a/src/src/detail/uri_resolve.cpp
+++ b/src/src/detail/uri_resolve.cpp
@@ -12,6 +12,7 @@
 #include <boost/algorithm/string/erase.hpp>
 #include <boost/algorithm/string/replace.hpp>
 #include <boost/algorithm/string/predicate.hpp>
+namespace network_boost = boost;
 #else   // NETWORK_URI_EXTERNAL_BOOST
 #include "../boost/algorithm/string/find.hpp"
 #include "../boost/algorithm/string/erase.hpp"
